packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -643,4 +647,124 @@ export class JobService {
 
     return true;
   }
+
+  public async getDetails(userId: number, jobId: number): Promise<JobDetailsDto> {
+    const jobEntity = await this.jobRepository.findOne({ id: jobId, userId });
+
+    if (!jobEntity) {
+        this.logger.log(ErrorJob.NotFound, JobService.name);
+        throw new NotFoundException(ErrorJob.NotFound);
+    }
+
+    const { chainId, escrowAddress, manifestUrl, manifestHash } = jobEntity;
+    const signer = this.web3Service.getSigner(chainId);
+
+    const escrowClient = await EscrowClient.build(signer);
+    const stakingClient = await StakingClient.build(signer);
+
+    const tokenAddress = await escrowClient.getTokenAddress(escrowAddress);
+
+    const [balance, allocation, manifestData, paidOut] = await Promise.all([
+      escrowClient.getBalance(escrowAddress),
+      stakingClient.getAllocation(escrowAddress),
+      this.getManifest(manifestUrl),
+      this.getPaidOutAmount(chainId, tokenAddress, escrowAddress)
+    ]);
+
+    if (!manifestData) {
+      throw new NotFoundException(ErrorJob.ManifestNotFound);
+    }
+
+    const manifest = (manifestData as FortuneManifestDto).requestType === JobRequestType.FORTUNE
+           ? manifestData as FortuneManifestDto
+           : manifestData as CvatManifestDto;
+
+    const exchangeOracleAddress = (manifest as FortuneManifestDto).requestType === JobRequestType.FORTUNE 
+                         ? ConfigNames.FORTUNE_EXCHANGE_ORACLE_ADDRESS 
+                         : ConfigNames.CVAT_EXCHANGE_ORACLE_ADDRESS;
+
+    const recordingOracleAddress = this.configService.get<string>(ConfigNames.RECORDING_ORACLE_ADDRESS)!;
+    const reputationOracleAddress = this.configService.get<string>(ConfigNames.REPUTATION_ORACLE_ADDRESS)!;
+
+
+    let manifestDetails;
+ 
+    if ((manifest as FortuneManifestDto).requestType === JobRequestType.FORTUNE ) {
+      manifestDetails = {
+        chainId,
+        title: (manifest as FortuneManifestDto).requesterTitle,
+        description: (manifest as FortuneManifestDto).requesterDescription,
+        requestType: JobRequestType.FORTUNE,
+        submissionsRequired: (manifest as FortuneManifestDto).submissionsRequired,
+        tokenAddress,
+        fundAmount: (manifest as FortuneManifestDto).fundAmount,
+        requesterAddress: signer.address,
+        exchangeOracleAddress,
+        recordingOracleAddress,
+        reputationOracleAddress
+      }
+    } else {
+      manifestDetails = {
+        chainId,
+        requestType: (manifest as CvatManifestDto).annotation.type,
+        submissionsRequired: (manifest as CvatManifestDto).annotation.job_size,
+        tokenAddress,
+        fundAmount: Number((manifest as CvatManifestDto).job_bounty),
```

Comment:
`job_bounty` reflects an approx reward user will receive for annotating a job (part of the dataset)